Adds OmegaStation back into rotation.

sergeirocks100 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

This, well, readds OmegaStation back to the game. I've spent the past month updating it, most of the excluded areas and all of the missing non-head jobs have been added in, many of the issues with the map have been fixed, and i think it's more or less ready for rotation.

Here's a screenshot:[https://amel.pw/trevie/screenshots/s8.png](https://amel.pw/trevie/screenshots/s8.png)

Credits go to me, of course, Okand and whoever else contributed to the map, and the developers of Skyrat's version of Omega, for giving me ideas on where to put things as well as the Toxins Lab and Quartermaster's Office copypasted from their version of the map.

I highly recommend that this gets test merged first, as there might be bugs with the map that need to be ironed out.

## Why It's Good For The Game

Right now, i don't think /tg/ has any real lowpop maps. Pubby is the only "low-pop" map in rotation, and not only is it as large as a medium-pop map, it doesn't really serve low population counts any better than, for example, Meta does.

Omega is more tailored for low population counts than Pubby is, and i think it will do well on low population servers such as Campbell, small or private servers, or during graveyard hours on more populated servers.

## Changelog
:cl:
add: NT's decommissioned "Omega" line of stations have been put back into use.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

sergeirocks100 committed:
adds OmegaStation to the game

sergeirocks100 committed:
Merge remote-tracking branch 'upstream/master'

sergeirocks100 committed:
Update OmegaStation.dmm

sergeirocks100 committed:
Update OmegaStation.dmm

sergeirocks100 committed:
Update OmegaStation.dmm

sergeirocks100 committed:
Merge remote-tracking branch 'upstream/master'

sergeirocks100 committed:
Update OmegaStation.dmm

sergeirocks100 committed:
Merge remote-tracking branch 'upstream/master'

sergeirocks100 committed:
Update OmegaStation.dmm

sergeirocks100 committed:
Update OmegaStation.dmm

sergeirocks100 committed:
more work

sergeirocks100 committed:
Merge remote-tracking branch 'upstream/master'

sergeirocks100 committed:
Update OmegaStation.dmm

sergeirocks100 committed:
whyte ship

sergeirocks100 committed:
Merge remote-tracking branch 'upstream/master'

sergeirocks100 committed:
fixes

sergeirocks100 committed:
Update OmegaStation.dmm

sergeirocks100 committed:
Update OmegaStation.dmm

sergeirocks100 committed:
work, i guess

sergeirocks100 committed:
Merge remote-tracking branch 'upstream/master'

sergeirocks100 committed:
Delete OmegaStation2.dmm

sergeirocks100 committed:
job additions

sergeirocks100 committed:
Merge remote-tracking branch 'upstream/master'

sergeirocks100 committed:
more updates

sergeirocks100 committed:
more updates

sergeirocks100 committed:
Merge remote-tracking branch 'upstream/master'

sergeirocks100 committed:
another update (and possibly final before the PR)

sergeirocks100 committed:
forgot to remove the non-functional whiteship

sergeirocks100 committed:
Update OmegaStation.dmm

sergeirocks100 committed:
Merge remote-tracking branch 'upstream/master'

sergeirocks100 committed:
Update OmegaStation.dmm

sergeirocks100 committed:
Merge remote-tracking branch 'upstream/master'

sergeirocks100 committed:
Merge remote-tracking branch 'upstream/master'

sergeirocks100 committed:
final commit before the PR (hopefully)

sergeirocks100 committed:
Merge remote-tracking branch 'upstream/master'

Ghilker commented:
you have to remake the piping to layer2 and layer3 for distro and waste as all our maps have now

Rohesie commented:
Very nice

wesoda25 commented:
Will you be maintaining this map in the future?

Fikou commented:
missing the anomaly compressor

Fikou commented on _maps/map_files/OmegaStation/job_changes.dm:
```diff
@@ -0,0 +1,143 @@
+MAP_REMOVE_JOB(hos)
+MAP_REMOVE_JOB(chief_engineer)
+MAP_REMOVE_JOB(cmo)
+MAP_REMOVE_JOB(rd)
```
```suggestion
MAP_REMOVE_JOB(rd)

```
trailing newline

Fikou commented on code/modules/jobs/map_changes/map_changes.dm:
```diff
@@ -5,3 +5,6 @@
 #undef JOB_MODIFICATION_MAP_NAME
+
+#include "..\..\..\..\_maps\map_files\OmegaStation\job_changes.dm"
+#undef JOB_MODIFICATION_MAP_NAME
```
```suggestion
#undef JOB_MODIFICATION_MAP_NAME

```
ditto

kriskog commented:
You need to make sure it works with #46956, or look into having that fixed. You say you did the missing head roles, just make sure any other missing jobs are actually working. There used to be several issues related to omegastation and missing roles that were closed with its removal.

Fikou commented:
some rooms like the diner, toxins and experimentor feel really weird and empty
xenobio's assymetry is really unepic

ArcaneMusic commented:
You're merging straight from your master :eyes:

GravitronBlue commented:
From the screenshot the two right side xenobiology pens appear to lack disposal bins for the disposal outlets in them.

sergeirocks100 committed:
Update _maps/map_files/OmegaStation/job_changes.dm

Co-authored-by: Fikou <piotrbryla@onet.pl>

sergeirocks100 committed:
Update code/modules/jobs/map_changes/map_changes.dm

Co-authored-by: Fikou <piotrbryla@onet.pl>

sergeirocks100 commented:
Sorry for the lack of replies, i was staring at the inside of my own eyelids.

I plan on redoing the piping in the future, but it would have been a real pain in the ass to do, and the current piping does the job, so this is something i've put off for now.

I do plan on maintaining this map if it gets merged into the codebase.

I don't know what a "anomaly compressor" is, i'm not sure why i would need cytology components when, as far as i know, you can print those at the protolathe, i'll see what i can do about the lack of spawn points.

I had no problems with late joining during my testing of the map, so i think i'm fine.

I do agree that some areas look a little plain, but i don't really know how i would address that issue, i'll try to fix Xenobio's two lopsided pens as well as the disposal chute issues with them.

sergeirocks100 committed:
Update OmegaStation.dmm

Fikou commented:
/obj/machinery/research/explosive_compressor
place it in toxins

there should be one cytology locker somewhere in xenobio

sergeirocks100 committed:
Update OmegaStation.dmm

sergeirocks100 commented:
done, and done

sergeirocks100 committed:
windoors for the upload consoles

sergeirocks100 committed:
Update OmegaStation.dmm

FloranOtten commented:
YES! FINALLY! BLESSED MAP RETURNS

TheVekter commented:
Maybe extend Xenobio a little? Needs more room, more pens.

sergeirocks100 committed:
Update OmegaStation.dmm

sergeirocks100 commented:
I expanded it out to 7 pens, i probably won't be expanding it more after this, though.

uomo91 commented:
I got some runtimes on this, one related to radio stuff, one related to lattices.

uomo91 commented:
Also, any chance of a public mining shuttle getting added?

sergeirocks100 commented:
I'm not very experienced with SS13 mapping, but i'll see if i can do something about the runtimes.

And while a public mining shuttle may be added in sometime, right now, i have no clue where i would put it.

Rohesie commented:
By a headcoder decree, we cannot add more maps without removing existing ones.
A lowpop map is good, and with some changes and fixes this one looks promising.
IceBox or Pubby, which one goes?

TheVekter commented:
That seems odd, but y'all probably know something I don't.

Should put it to a vote I guess?

sergeirocks100 committed:
moisture traps

uomo91 commented:
![psych](https://user-images.githubusercontent.com/51800976/92418461-35673500-f12d-11ea-997d-ba54235c37bf.png)

Would you consider redoing the seats and table in the Psych's office like this?
Only with the couch on the north wall is it possible for patients to lay on the couch, which I thought was really funny.

sergeirocks100 commented:
I think that we should put it to a vote as well. Although if i got to pick, i would probably pick Pubby. Ice Box is too much of a interesting idea to remove, and Pubby, as much as i like it, isn't all that great for lowpop in my opinion.

I'll go ahead and rearrange the Psychologist's Office's couches.

sergeirocks100 committed:
psychologist's office rearrangment

uomo91 commented:
@sergeirocks100 I've fixed all the runtimes I've found, I'll PR to your branch shortly. Merge it if you'd like.

There were some doubled up catwalks on the solar, the AI intercomms had some funky var probably left over from oldcode or something, and some APCs were misnamed and had incorrect area strings.

sergeirocks100 commented:
Thanks.

uomo91 committed:
Fixes runtimes.

uomo91 commented:
https://github.com/sergeirocks100/tgstation/pull/1

sergeirocks100 committed:
Merge pull request #1 from uomo91/omega

Fixes runtimes.

wesoda25 commented:
Fact of the matter is there’s a reason pubby is in the game and omega isn’t

uomo91 committed:
Fixes and tweaks solars.

LemonInTheDark commented:
Mapmerge the shuttle yeah?

uomo91 commented:
@sergeirocks100 I opened another PR, I fixed up the solar (no longer roundstart wired, has some spaces added so wiring doesn't look goofy), solar maint power stuff, and removed the rapid pipe cleaner layers from engineering since they're not useful and were taken out of other maps.

uomo91 committed:
Wire fix for engineering foyer APC.

sergeirocks100 commented:
Again, thank you.

sergeirocks100 committed:
Merge pull request #2 from uomo91/omega

Solar fixes and tweaks.

uomo91 commented:
No problem.
I'm noticing some pretty big sections of wiring that are messed up or missing, look at nanite and experimentor rooms for instance.
Port maint seems to totally lack an APC.

sergeirocks100 commented:
To be honest, i think i forgot to add wiring to that area, i'll do that right away.

uomo91 commented:
I'm also seeing some disconnected areas sharing area types, mostly maint ones.
You shouldn't do that, give each physically distinct area its own APC.

I'll try to give fixing more of this stuff a go tomorrow maybe.

sergeirocks100 committed:
Update OmegaStation.dmm

sergeirocks100 committed:
Update OmegaStation.dmm

cacogen commented:
I hate more

cacogen commented:
We don't need more maps. We need less maps that are maintained and iterated on to a higher standard.

sergeirocks100 committed:
Update OmegaStation.dmm

ShadowmanV92 commented:
Nobody likes pubby, all my homies hate pubby. Omega is at least a breathe of fresh air and much more of a lowpop map then pubby ever was

uomo91 committed:
Fixing a few things.

Fixes AI sat wiring, duplicate APC in prison, random cable in engineering under door, misplaced virologist landmark.

uomo91 committed:
Powernet stuff.

Connects engineering SMES to power grid, fixes gravity generator wiring, fixes some wiring and APC dir's in science, fixes turbine wiring, attempts to make the station's wiring layout more robust and with fewer single points of failure.

uomo91 commented:
I just opened a new PR that fixed a lot of powernet things.

- Fixed AI sat SMES and wiring.
- Got rid of a duplicate APC in prison.
- Two random misplaced power cables deleted.
- There was a virologist spawn landmark in the SM, yikes. No longer.
- Properly connected the engineering SMES to the grid.
- Tried to add more redundancy to the station's wiring layout so there aren't a bunch of huge points of failure that wipe out half the station's power with one wire snip.
- Fixed some more wiring and APC placement in sci and maint.
- Properly set up the turbine and connected it to a SMES and the grid.
- Same thing with the powernet fixes, for the gravgen.
- Cleaned up a couple assorted small things I found like item placement in maint.

sergeirocks100 committed:
Merge pull request #3 from uomo91/omega

A lot of powernet stuff and some other misc things.

uomo91 commented:
Also eventually, probably sooner rather than later, you're going to need to turn this into TGM format.
Look into the map editor called StrongDMM, if you're not already using it.

sergeirocks100 commented:
Again, thank you for helping me out with my map.

I have been using StrongDMM for the map, but i don't know how to export to TGM.

uomo91 commented:
> Again, thank you for helping me out with my map.

Very happy to help, I love Okand's work.

> I have been using StrongDMM for the map, but i don't know how to export to TGM.

File > Preferences > Save Options > Map Save Format > then switch from provided or Byond, to TGM.

uomo91 committed:
Fixes more stuff.

Fixes SM wiring so it's isolate from the power grid, adds missing plating to the windows by the solar exit,  improves placement of gravgen power equipment.

uomo91 commented:
I opened another PR mostly fixing things I messed up.
The wiring and SMES placement in the gravgen is better now, the SM is isolated from the main powernet now so it won't fry anyone who gets shocked, and some missing plating under the windows near the solars was added.

sergeirocks100 committed:
Merge pull request #4 from uomo91/omega

Fixes even more stuff.

uomo91 commented:
Atmos seems to lack a mixer, and the turbine has a "incinerator chamber output inlet" instead of a scrubber inside... I don't play a lot of atmos so I'm not sure what to do to remedy the above, if anything.

Kylerace committed:
Adds more canisters to toxins

it is now 4 o2 canisters 4 plasma

Kylerace commented:
toxins doesnt have enough canisters, so i made a pr to your branch to make it have 4 o2 canisters and 4 plasma by making the toxins storage room slightly bigger. i like the layout of this station but i have some concerns. i dont like how atmospherics is the same size as the rest of engineering combined. in fact the atmospherics room in this station is even bigger than metas' atmos room, despite meta being about twice the size of this map. im not complaining that atmos is huge, but it doesnt make sense for it to be this size compared to everything else. honestly the bar takes up a ton of space too. i think that itd be best if vault and the ai upload were split up so that upload could have turrets, i think you could fit one of them in this little maint space.
![Screenshot_211](https://user-images.githubusercontent.com/15794172/92551749-a3862780-f213-11ea-9a22-be1993b926de.png)
there should probably be another solar on the opposite end of the station although i dont think its a big problem, and for some reason there seems to be a beaker of sulfuric acid in science on the circuit fabricator.

however a possibly big flaw i see so far with the station is that its scrubber and air supply pipes arent in different layers like the rest of the maps. uh, im not 100% sure if that will actually end up being a problem later if a maintainer refuses to merge it because of it. Anyways, i wish you best of luck and hopefully you can put icebox or pubby in the fucking garbage.

sergeirocks100 committed:
Update OmegaStation.dmm

sergeirocks100 committed:
Merge pull request #5 from Kylerace/biggertoxinssergei

Adds more canisters to toxins

sergeirocks100 commented:
Thanks for helping with the map.

I'm not sure what i can do about the oversized atmos and the upload, but i'll look into it.

Omega long predates the great repiping, i do plan on repiping it one day if it gets merged. But it would be a pain in the ass to replace them all right now, and the existing pipes do the job just fine, so i'm leaving it alone for now.

sergeirocks100 committed:
Update OmegaStation.dmm

sergeirocks100 committed:
Update OmegaStation.dmm

81Denton commented:
@tgstation/commit-access What's the consensus on this considering there was a recent vote that pruned the amount of maps we have? I don't want @sergeirocks100 to waste his time.

Rohesie commented:
This would have to be a replacement for an existent map ([source](https://sb.atlantaned.space/info/mapularity)).
Currently Pubby is the least played map, and I'd be up for making the switch to Omega, for variety if nothing else.

Jared-Fogle commented:
Agreed with @Rohesie--if Pubby is meant to be the low pop map, then Omega should replace it.

81Denton commented:
I'm more partial towards Pubby because it doesn't suffer from the explosion issue that Omega suffers from (maxcaps can make entire areas uninhabitable due to its small size). I think that adding Omega for "extreme" lowpop with a very low pop limit on top of the existing maps would work well too.

Jared-Fogle commented:
I'm personally opposed to adding a new map without replacing another one just for the sake of maintainability.

Rohesie commented:
I'd rather we had as many maps as decent and maintained versions exist, and this would imply keeping the existing 4 maps and adding OmegaStation and ShipStation on top of them. I don't feel like 6 maps is unreasonable, and each offers something different.

That said, oranges wants us to move towards less maps, ideally two, so we have to make choices.
I'd rather we saw variety and rotation and were not forever stuck on the same maps, myself. I like Pubby and feel it can return in t he future, but I'd favor change (when the incoming map is good) over just staying as we are.

optimumtact commented:
The consensus is that my decree is the same as it has been for some time.

If you want new maps in the codebase you need to replace an existing one.

EOBGames commented:
Hello! I've run your map over with the debug tools to check for any issues. Here's what I've found:
Camera Debugger:
![cameradebugoutput](https://user-images.githubusercontent.com/58124831/92663304-2e911b80-f301-11ea-946e-a9e43b26fea2.PNG) You have a few matching c-tags, this will cause issues for viewing the cameras via a console. Resolve these by editing the c-tag variable on the offending cameras to something unique.

Atmos Debugger:
![pipenetdebugouput](https://user-images.githubusercontent.com/58124831/92663388-67c98b80-f301-11ea-94db-2d2ea2170813.PNG)
There are quite a few disconnected atmospherics pipes, mostly in the brig near the entrance to permabrig. This is an easy fix, just connect up the missing pipes here:
![image](https://user-images.githubusercontent.com/58124831/92663492-bd059d00-f301-11ea-8650-807ba77073e7.png)
and here:
![image](https://user-images.githubusercontent.com/58124831/92663577-f807d080-f301-11ea-9716-f7fa122ea06a.png)
Additionally, an air vent is disconnected here, in toxins mixing:
![image](https://user-images.githubusercontent.com/58124831/92663626-0f46be00-f302-11ea-9b40-adc70a112047.png)
On a personal note for atmos, I believe the map could benefit from a 2-layered atmospherics system, as is now more standard on our current maps. This would reduce the number of pipes in walls (which causes issues for repairs) and generally simplify the layout of the pipenet.

Powernet debugger:
![powernetdebugoutput](https://user-images.githubusercontent.com/58124831/92663716-4f0da580-f302-11ea-9778-b41ed182d8a1.PNG)
Not much here! This cable in the brig is doubled, please remove one of the cables:
![image](https://user-images.githubusercontent.com/58124831/92663780-82e8cb00-f302-11ea-9973-48e17a0c7ecd.png)
The AI satellite being unwired is, I believe, intentional.

There are several doubled APCs: the fore primary hallway, aft primary hallway, and starboard primary hallway all have two APCs each. The port maint also has two, one below virology, and one to the left of the library. Remove one or the other.

The camera coverage in the far east of science is rather spotty, particularly in the hallway between genetics and the servers. I would strongly recommend adding a camera somewhere around the door to genetics.

Hydroponics has no APC. Unless you want sad botanists, I'd add one.

Permabrig also has no APC. Unless you want a riot on your hands, I'd add one.

Toxins Storage has no APC, again I'd add one- generally all station areas should have an APC. This area is also lacking an air alarm, which should also probably be added.

Tech storage has no air alarm.

Several rooms to the right side of science have no intercoms, particularly genetics.

The Medbay APC is in an area that is inaccessible to anyone without medbay access- this isn't ideal for engineers fixing it. Perhaps shift it elsewhere in the room, such as next to the door to virology?

Personal observation: I notice that the map has no disposals system. Adding this in could be a good goal to improve the map and help further advance it from the original version, if you're feeling up to it.

For future reference, the debug tools can be found via activating debug verbs in the debug tab, then within the mapping tab that appears. These really help for finding hard to spot problems with your map, such as the ones listed above.

sergeirocks100 commented:
Thanks for informing me of these issues, i'll try to fix them right away.

sergeirocks100 committed:
Update OmegaStation.dmm

sergeirocks100 committed:
Update OmegaStation.dmm

sergeirocks100 committed:
Merge remote-tracking branch 'upstream/master'

Cyberboss commented:
> The more advanced version of this is a seperate repo, which allows map creators to completely skip the the review cycle and gives them more freedom, but reduces the chances they will get normal contributor help updating their map.
> 
> Until then the process is unlikely to change unless there are new headcoders.

I'm fully in agreement with this. Which is why it's been a longstanding goal of mine to push out support for maps that can be remotely downloaded with a versioning system, pushing that tax onto people that want their map to be played.

Unfortunately, life has been getting in the way of me setting it up.

Cyberboss commented:
I have the overall steps outlined for that. If anyone wants to help, ping me on discord.

SpaceManiac commented:
> I have the overall steps outlined for that. If anyone wants to help, ping me on discord.

Is writing them down that hard

sergeirocks100 committed:
Update OmegaStation.dmm

sergeirocks100 committed:
Update OmegaStation.dmm

sergeirocks100 committed:
Update OmegaStation.dmm

sergeirocks100 committed:
Update OmegaStation.dmm

sergeirocks100 committed:
new maint theatre from Skyrat

sergeirocks100 committed:
maint bar has alcohol now

sergeirocks100 committed:
kitchen improvements

Cyberboss commented:
Started some work in #53663 basically the last thing I need is a way to download and unzip .zips from GitHub via rust-g. I asked @spookydonut to potentially help with that

sergeirocks100 committed:
converts the whiteship to TGM (and hopefully makes it pass Travis)

sergeirocks100 committed:
Update OmegaStation.dmm

sergeirocks100 committed:
Merge remote-tracking branch 'upstream/master'

Jared-Fogle commented:
White ships are randomized now--please don't add another extremely small white ship since they are no longer map specific.

sergeirocks100 commented:
I realized that too, which is why i will probably be removing Omega's whiteship dock for now, as there isn't a single white ship in the game right now besides Omega's own that can fit in the tiny space.

sergeirocks100 committed:
removes Omega's whiteship and its dock

sergeirocks100 committed:
improves the testing range

sergeirocks100 committed:
Update OmegaStation.dmm

sergeirocks100 committed:
Merge remote-tracking branch 'upstream/master'

sergeirocks100 committed:
Update OmegaStation.dmm

actions-user committed:
Automatic changelog compile [ci skip]

optimumtact commented on html/changelog.html:
```diff
@@ -51,6 +51,25 @@
+				<li class="bugfix">Apple Cakes no longer slice into Slime Cake Slices</li>
+				<li class="bugfix">Minor runtime with no gameplay impact related to printing implant cases.</li>
+				<li class="bugfix">Non-gameplay impacting runtime on holopads when the user spoke after placing a call but before it has been connected.</li>
+				<li class="bugfix">EMP-shielded cameras will no longer runtime when destroyed.</li>
```
makes you cringe

SpaceManiac commented:
I'm closing this for now in honor of our standing policy against new maps in the codebase.

I recommend keeping track of or getting involved in the project to support out-of-codebase maps.

You should avoid making PRs from your fork's "master" branch, as our changelog cronjobs will run there and cause conflicts.

TheVekter commented:
I... okay? What process needs to be followed to replace one of the maps currently implemented in the codebase?

sergeirocks100 commented:
As saddened as i may be at this PR getting closed, i understand your point.

I don't like to make a new branch for every PR, as i have to download over 1.5 gigabytes worth of stuff every time i make a fork, and that's pretty unbearable on my slow DSL connection.

I had no idea that there was a policy against adding in new maps period, and if i had known this from the start, i probably wouldn't have bothered with this at all.

I will still be keeping this map up to date, however, and maybe a few months from now, i'll PR it again.

SpaceManiac commented:
I like OmegaStation as a map, but our team's circumstances are such that we don't want to maintain more than the 4 maps we already are. Maybe this will change in the future.

> I don't like to make a new branch for every PR, as i have to download over 1.5 gigabytes worth of stuff every time i make a fork, and that's pretty unbearable on my slow DSL connection.

Redownloading everything is absolutely not necessary to use a different branch name than `master` for your PRs. Our [Setting up git](https://tgstation13.org/wiki/Setting_up_git) guide has some details, and you can read the [Git manual](https://git-scm.com/doc) for more.

In particular the section describing `git checkout -b <BRANCHNAME>` is relevant.

sergeirocks100 commented:
You do know that Omega was going to replace another map, right?

SpaceManiac commented:
"going to replace another map" is not a path to mergeability because it does not concretely specify _which_ other map would be replaced.

After discussion with @Timberpoes, you are welcome to open a new PR if you both:

1. Title the PR something like "Replace PubbyStation with OmegaStation", since that's what would have to happen for the PR to be merged.
    - It's up to you whether you want to remove PubbyStation in the PR itself or leave that step to us (maintainers) to handle.
2. Use a branch name other than `master`. This is important for technical reasons and, as I said before, *does not* require multiple redownloads of the repository.

If you're having trouble managing the branches, ask for help on coderbus.

sergeirocks100 commented:
Thank you, i'm going to go ahead and do just that after i manage to fix up my repository.
